### PR TITLE
Add API to create hierarchical errors at java level

### LINKED ISSUE #28734: No API to create hierarchical errors at java level
**Description:**
```ballerina
// Level 1
public type Error distinct error;

// Level 2
public type ClientError distinct Error;

// Level 3
public type GenericClientError distinct ClientError;
```
The requirement is to create `http:GenericClientError` from java level and return. The ballerina API returns `http:ClientError`. Theoretically this should work as `http:GenericClientError` is a child error type of `http:ClientError`.

But the exposed API[1] is only creating level 1 errors. Therefore the created error type is `error`
Selection_210

This leads to a type incompatibility.
`incompatible types: 'error' cannot be cast to 'http:Response|http:ClientError'`

 [1] https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java#L150

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose

Fixes #28734 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java [hunk 1]
```diff
@@ -139,14 +139,34 @@ public static BError createError(Throwable error) {
...
+    }
+
     /**
      * Create an distinct error with given typeID, typeIdPkg and reason.
      *
      * @param typeIdName type id
      * @param typeIdPkg  type id module
      * @param message  error message
      * @return new error
+     * @deprecated Use @createError(module, errorTypeName, message, cause, details) to create a distinct error.
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java:167]: Previously we were able to create errors without defining the type in the bal. We actually create new type every time which is a bug. And this is also not similar to create a record or object, where we need to define a type in the bal. I will remove these once stdlib usages  are migrated.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java [hunk 2]
```diff
@@ -139,14 +139,34 @@ public static BError createError(Throwable error) {
...
+     * @param message       error message
+     * @param cause         error cause
+     * @param details       error details
+     * @return error value
+     * @throws BError if given error type is not defined in the ballerina module.
+     */
+    public static BError createError(Module module, String errorTypeName,
+                                     BString message, BError cause, Object details) {
+        io.ballerina.runtime.internal.values.ValueCreator
+                valueCreator = io.ballerina.runtime.internal.values.ValueCreator.getValueCreator(module.toString());
```
Comment (riyafa) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java:156]: Why do we have to use the full package name here?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java [hunk 3]
```diff
@@ -509,19 +510,23 @@ void generateValueCreatorMethods(ClassWriter cw, List<BIRTypeDefinition> typeDef
...
         List<BIRTypeDefinition> objectTypeDefs = new ArrayList<>();
+        List<BIRTypeDefinition> errorTypeDefs = new ArrayList<>();
 
         for (BIRTypeDefinition optionalTypeDef : typeDefs) {
             BType bType = optionalTypeDef.type;
             if (bType.tag == TypeTags.RECORD) {
                 recordTypeDefSet.add(optionalTypeDef);
             } else if (bType.tag == TypeTags.OBJECT && Symbols.isFlagOn(bType.tsymbol.flags, Flags.CLASS)) {
                 objectTypeDefs.add(optionalTypeDef);
+            } else   if (bType.tag == TypeTags.ERROR) {
```
Comment (riyafa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:521]: Shall we fix the indentation?

File: tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/values/modules/errors/testErrors.bal [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import ballerina/jballerina.java;
+
+public type GenericError distinct error;
+
+public type UserError distinct GenericError;
+
+public function getError(string errorName) returns error = @java:Method {
+    'class: "org.ballerinalang.nativeimpl.jvm.tests.RuntimeAPIMethods"
+} external;
```
Comment (riyafa) [inline tests/jballerina-unit-test/src/test/resources/test-src/runtime/api/values/modules/errors/testErrors.bal:25]: Missing new line

**Review by riyafa — APPROVED** [review]

**Pull request merged**